core/templates/dev/head/filters.js

```diff
@@ -336,4 +336,34 @@ oppia.filter('summarizeDefaultOutcome', ['$filter', function($filter) {
     }
     return summary;
   }
-}])
+}]);
+
+oppia.filter('appendMetrixPrefix', [function() {
+  return function (input) {
+    return Math.abs(Number(input)) >= 1.0e+9
+
+    // Nine Zeroes for Billions
+    ? (Math.abs(Number(input)) / 1.0e+9).toFixed(1) + "B"
+
+    // Six Zeroes for Millions
+    : Math.abs(Number(input)) >= 1.0e+6
+
+    ? (Math.abs(Number(input)) / 1.0e+6).toFixed(1) + "M"
+    // Three Zeroes for Thousands
+    : Math.abs(Number(input)) >= 1.0e+3
+
+    ? (Math.abs(Number(input)) / 1.0e+3).toFixed(1) + "K"
+
+    : Math.abs(Number(input));
+  };
+}]);
+
+oppia.filter('formattedObjective', [function() {
```

Comment:
I think this should be renamed to truncateAndCapitalize, so that it's more generic. At this point the filter doesn't really care that what's being passed to it is an objective.